src/NServiceBus.Transport.RabbitMQ/obsoletes-v8.cs

```diff
@@ -165,6 +167,303 @@ public static class RabbitMQTransportSettingsExtensions
+            where T : RabbitMQTransport
+        {
+            var transport = new RabbitMQTransport();
+
+            var routing = config.UseTransport(transport);
+
+            var settings = new RabbitMqTransportLegacySettings(transport, routing);
+
+            return settings;
+        }
+    }
+
+    public partial class RabbitMQTransport
+    {
+        internal string LegacyApiConnectionString { get; set; }
+        bool legacyMode;
+
+        internal RabbitMQTransport()
+            : base(TransportTransactionMode.ReceiveOnly, true, true, true)
+        {
+            legacyMode = true;
+        }
+
+        void ValidateAndApplyLegacyConfiguration()
+        {
+            if (!legacyMode)
+            {
+                return;
+            }
+
+            if (RoutingTopology == null)
+            {
+                throw new Exception("When using legacy API a routing topology must be configured with one of the 'EndpointConfiguration.UseTransport<RabbitMQTransport>().UseXXXXRoutingTopology()` methods.");
+            }
+
+            if (string.IsNullOrEmpty(LegacyApiConnectionString))
+            {
+                throw new Exception("When using legacy API a connection string must be configured with 'EndpointConfiguration.UseTransport<RabbitMQTransport>().ConnectionString()` method.");
+            }
+
+            if (LegacyApiConnectionString.StartsWith("amqp", StringComparison.OrdinalIgnoreCase))
+            {
+                AmqpConnectionString.Parse(LegacyApiConnectionString)(this);
+            }
+            else
+            {
+                NServiceBusConnectionString.Parse(LegacyApiConnectionString)(this);
+            }
+        }
+    }
+
+    /// <summary>
+    /// RabbitMQ transport configuration settings.
+    /// </summary>
+    public class RabbitMqTransportLegacySettings : TransportSettings<RabbitMQTransport>
+    {
+        internal RabbitMQTransport RabbitMqTransport { get; } //for testing the shim
+
+        internal RabbitMqTransportLegacySettings(RabbitMQTransport transport, RoutingSettings<RabbitMQTransport> routing)
+            : base(transport, routing)
+
+        {
+            RabbitMqTransport = transport;
+        }
+
+        [ObsoleteEx(
+            Message = "Using custom topologies is not possible in the legacy API. Use the new API.",
+            TreatAsErrorFromVersion = "7",
+            RemoveInVersion = "8")]
+        public RabbitMqTransportLegacySettings UseCustomRoutingTopology(
+            Func<bool, IRoutingTopology> topologyFactory)
+        {
+            throw new NotImplementedException();
+        }
+
+        [ObsoleteEx(
+            Message = "Disabling the durable exchanges is not possible in the legacy API. Use the new API and create the topology instance with appropriate arguments.",
+            TreatAsErrorFromVersion = "7",
+            RemoveInVersion = "8")]
```

Comment:
I thought that we always remove in V_error+1. Since this code will have to be removed to compile the solution against V7, I don't see a reason to keep the obsolete for one additional version.

Reply:
yeah, technically it's totally valid to remove at v_error+1. I was just wondering whether there is any cost of just keeping it with all the other APIs that are being removed anyway. But it shouldn't really matter too much 👍